tests/components/diagnostics/__init__.py

```diff
@@ -22,3 +23,28 @@ async def get_diagnostics_for_config_entry(hass, hass_client, domain_or_config_e
     )
     assert response.status == HTTPStatus.OK
     return await response.json()
+
+
+async def get_diagnostics_for_device(
+    hass, hass_client, domain_or_config_entry, device_id=None
+):
+    """Return the diagnostics for the specified device."""
+    if isinstance(domain_or_config_entry, str):
+        config_entry = MockConfigEntry(domain=domain_or_config_entry)
+        config_entry.add_to_hass(hass)
+    else:
+        config_entry = domain_or_config_entry
+
+    dev_reg = async_get(hass)
+    if device_id is None:
+        device = dev_reg.async_get_or_create(
```

Comment:
Are we creating a device here? How will that be used?

Reply:
The test needs a config entry and a device to be able to reach the platforms `async_get_device_diagnostics` method. Else error handling will return a 404

Reply:
Right, but I think that these helpers are now too broad for their main use case which will be other integrations testing their returned state. Those integrations should test the diagnostics by setting up the integration as normal and using this test helper. I think for testing the core diagnostics integration here we don't need this flexibility but can create the mock device explicitly.

Reply:
even for the first helper, the right way to do this for testing other integrations would be to provide the config entry right? So the domain logic only helps for the diagnostics tests, same with not providing a device ID. I think we may as well update both helpers if we want to address that conern

Reply:
Yes, that's what I'm thinking too.

Reply:
If we want we can add an internal helper just for this integration.

Reply:
I don't think that's needed until we add more features that require different sets of tests. As long as we just add new types of diagnostics, we can do it all in the one test without having to create extra helpers